examples/vae/train_vae.py

```diff
@@ -3,154 +3,120 @@
-if args.gpu >= 0:
-    cuda.get_device_from_id(args.gpu).use()
-    model.to_gpu()
-xp = np if args.gpu < 0 else cuda.cupy
-
-# Setup optimizer
-optimizer = optimizers.Adam()
-optimizer.setup(model)
-
-# Init/Resume
-if args.initmodel:
-    print('Load model from', args.initmodel)
-    serializers.load_npz(args.initmodel, model)
-if args.resume:
-    print('Load optimizer state from', args.resume)
-    serializers.load_npz(args.resume, optimizer)
-
-# Learning loop
-for epoch in six.moves.range(1, n_epoch + 1):
-    print('epoch', epoch)
-
-    # training
-    perm = np.random.permutation(N)
-    sum_loss = 0       # total loss
-    sum_rec_loss = 0   # reconstruction loss
-    for i in six.moves.range(0, N, batchsize):
-        x = chainer.Variable(xp.asarray(x_train[perm[i:i + batchsize]]))
-        optimizer.update(model.get_loss_func(), x)
-        if epoch == 1 and i == 0:
-            with open('graph.dot', 'w') as o:
-                g = computational_graph.build_computational_graph(
-                    (model.loss, ))
-                o.write(g.dump())
-            print('graph generated')
-
-        sum_loss += float(model.loss.data) * len(x.data)
-        sum_rec_loss += float(model.rec_loss.data) * len(x.data)
-
-    print('train mean loss={}, mean reconstruction loss={}'
-          .format(sum_loss / N, sum_rec_loss / N))
-
-    # evaluation
-    sum_loss = 0
-    sum_rec_loss = 0
+import matplotlib
+matplotlib.use('Agg')
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Chainer example: VAE')
+    parser.add_argument('--initmodel', '-m', default='',
+                        help='Initialize the model from given file')
+    parser.add_argument('--resume', '-r', default='',
+                        help='Resume the optimization from snapshot')
+    parser.add_argument('--gpu', '-g', default=-1, type=int,
+                        help='GPU ID (negative value indicates CPU)')
+    parser.add_argument('--out', '-o', default='result',
+                        help='Directory to output the result')
+    parser.add_argument('--epoch', '-e', default=100, type=int,
+                        help='number of epochs to learn')
+    parser.add_argument('--dimz', '-z', default=20, type=int,
+                        help='dimention of encoded vector')
+    parser.add_argument('--batchsize', '-b', type=int, default=100,
+                        help='learning minibatch size')
+    parser.add_argument('--test', action='store_true',
+                        help='Use tiny datasets for quick tests')
+    args = parser.parse_args()
+
+    print('GPU: {}'.format(args.gpu))
+    print('# dim z: {}'.format(args.dimz))
+    print('# Minibatch-size: {}'.format(args.batchsize))
+    print('# epoch: {}'.format(args.epoch))
+    print('')
+
+    # Prepare VAE model, defined in net.py
+    model = net.VAE(784, args.dimz, 500)
+    if args.gpu >= 0:
+        chainer.cuda.get_device_from_id(args.gpu).use()
+        model.to_gpu()
```

Comment:
This example is using `StandardUpdater` with `device=args.gpu` argument, and the updater automatically transfer the model to the specified GPU in its constructer (https://github.com/chainer/chainer/blob/master/chainer/training/updater.py#L144-L146), so these part (L:47-49) is actually not needed. So how about removing these lines?